.github/workflows/github_pages.yml

```diff
@@ -85,10 +85,28 @@ jobs:
           chmod -c -R +rX "${{ inputs.output-path }}" | while read line; do
             echo "::warning title=Invalid file permissions automatically fixed::$line"
           done
+      - name: Archive artifact
+        shell: sh
+        run: |
+          echo "::group::Archive artifact"
+          tar \
+            --dereference \
+            --hard-dereference \
+            --directory "$OUTPUT_PATH" \
+            -cvf "$RUNNER_TEMP/artifact.tar" \
+            --exclude=.git \
+            --exclude=.github \
+            .
+          echo "::endgroup::"
+        env:
+          OUTPUT_PATH: ${{ inputs.output-path }}
```

Comment:
This is exactly the same as the ["Archive artifact" step from upload-pages-artifact@v3.0.1](https://github.com/actions/upload-pages-artifact/blob/56afc609e74202658d3ffba0e8f6dda462b719fa/action.yml#L24-L38) other than `INPUT_PATH` being renamed to `OUTPUT_PATH` to match the name used elsewhere in our file.

In v4.0.0 upload-pages-artifact just added `--exclude=".[^/]*"` to [the `tar` command](https://github.com/actions/upload-pages-artifact/blob/7b1f4a764d45c48632c6b24a0339c27f5614fb0b/action.yml#L24-L39) to exclude hidden file.